UsefulProteomicsDatabases/ProteinDbRetriever.cs

```diff
@@ -0,0 +1,169 @@
+        /// <param name="absolutePathToStorageDirectory"></param>
+        public static void RetrieveProteome(string proteomeID, string absolutePathToStorageDirectory, ProteomeFormat format, Reviewed reviewed, Compress compress, Include include)
+        {
+            if (Directory.Exists(absolutePathToStorageDirectory))
+            {
+                string htmlQueryString = "";
+                string filename = "\\" + proteomeID;
+                if (format == ProteomeFormat.fasta)
+                {
+                    if(reviewed == Reviewed.yes)
+                    {
+                        filename += "_reviewed";
+                    }
+                    else
+                    {
+                        filename += "_unreviewed";
+                    }
+                    //only fasta style proteome allows retrieval of extra isoforms
+                    if(include == Include.yes)
+                    {
+                        filename += "_isoform";
+                    }
+                    filename += ".fasta";
+                    if (compress == Compress.yes)
+                    {
+                        filename += ".gz";
+                    }
+                    htmlQueryString = "https://www.uniprot.org/uniprot/?query=proteome:" + proteomeID + " reviewed:" + reviewed + "&compress=" + compress + "&format=" + format + "&include:" + include;
+                }
+                else if (format == ProteomeFormat.xml)
+                {
+                    if (reviewed == Reviewed.yes)
+                    {
+                        filename += "_reviewed";
+                    }
+                    else
+                    {
+                        filename += "_unreviewed";
+                    }
+                    filename += ".xml";
+                    if (compress == Compress.yes)
+                    {
+                        filename += ".gz";
+                    }
+                    htmlQueryString = "https://www.uniprot.org/uniprot/?query=proteome:" + proteomeID + " reviewed:" + reviewed + "&compress=" + compress + "&format=" + format;
+                }
+                if(htmlQueryString.Length > 0)
+                {
+                    using (WebClient Client = new WebClient())
+                    {
+                        Client.DownloadFile(htmlQueryString, absolutePathToStorageDirectory + filename);
+                    }
+                }                
+            }
+        }
+
+        /// <summary>
+        /// The list of available UniProt Proteomes was downloaded 09/03/2020
+        /// This is returned as a dictionary
+        /// key = Proteome ID
+        /// value = Organism
+        /// </summary>
+        /// <returns></returns>
+        public static Dictionary<string,string> UniprotProteomesList()
+        {
+            string currentDirectory = Directory.GetCurrentDirectory();
+            string filePath = Path.Combine(currentDirectory, "UniProtProteomes.txt");
```

Comment:
I believe the answer is yes, and i have added a skeleton for that. For instance, we can request different file formats (text/tab, etc.). And there is a list of known columns. So we can submit protein list and get back specific metadata.

However, there is also a way to search with key word (see below). We just have to format the query correctly. And, i'd need a constructor for that. If you have an example in mind, i'll try to make the constructor and a unit test. Otherwise, i prefer just ot let this dog lie. We really have immediate use for proteomes and no specific use for other types of queries currently

https://www.uniprot.org/help/text-search

Reply:
Sounds good. The reason I ask is that we have to duplicate the text files in mzLib currently, e.g. https://github.com/smith-chem-wisc/mzLib/blob/master/Proteomics/ProteolyticDigestion/proteases.tsv and https://github.com/smith-chem-wisc/MetaMorpheus/blob/master/EngineLayer/ProteolyticDigestion/proteases.tsv. Maybe that's been resolved, but at the time, we had trouble getting the nuget file to package the proteases.tsv file, so the same may be true for these text files.

Reply:
On that note, these text files should be added to the [nuspec file](https://github.com/smith-chem-wisc/mzLib/blob/master/mzLib.nuspec) if you do end up keeping them.